plugin-test/Categorifier/Test/HList.hs

```diff
@@ -59,3 +66,38 @@ zipWith fn (HCons1 hf tf) (HCons1 hg tg) = HCons1 (fn hf hg) $ zipWith fn tf tg
 zipLowerWith :: (forall a. f a -> g a -> c) -> HList1 f l -> HList1 g l -> Vec (Length l) c
 zipLowerWith _ HNil1 HNil1 = VNil
 zipLowerWith fn (HCons1 hf tf) (HCons1 hg tg) = fn hf hg ::: zipLowerWith fn tf tg
+
+data HMap1 (f :: v -> Type) (l :: [(k, v)]) where
+  HEmpty1 :: HMap1 f '[]
+  HInsert1 :: Proxy k -> f e -> HMap1 f l -> HMap1 f ('(k, e) ': l)
+
+type family Value l k :: Type where
+  Value '[] k = Void
```

Comment:
Is it better to use `TypeError "message"` instead of `Void` here?

Reply:
Probably.

But this wasn't actually meant to be committed yet. It's part of some better HMap facilities that I haven't gotten working yet. Figured it was better to have this do _something_ (which is make the correlation between test exprs and test cases more obvious) rather than get a fully-featured HMap together.